events/src/main/kotlin/edu/artic/events/AllEventsAdapter.kt

```diff
@@ -12,10 +14,15 @@ import kotlinx.android.synthetic.main.cell_all_events_layout.view.*
  * @author Sameer Dhakal (Fuzz)
  */
 
-class AllEventsAdapter() : AutoHolderRecyclerViewAdapter<AllEventsCellViewModel>() {
+class AllEventsAdapter(val viewModel: AllEventsViewModel) : AutoHolderRecyclerViewAdapter<AllEventsCellViewModel>() {
 
     override fun View.onBindView(item: AllEventsCellViewModel, position: Int) {
 
+        clicks()
```

Comment:
theres already a `onClickListener` that only ever gets set once on created view but does the proper item lookup for you. can we just use that directly instead of recreating the binding at bind time?

Reply:
there's even a `itemClicks()` extension somewhere on `adapter` that you can hook up to your viewmodel on the fragment level (instead of within the adapter)